src/Hipace.cpp

```diff
@@ -579,12 +475,28 @@ Hipace::SolveOneSlice (int islice, const int islice_local, int step)
     // Psi ExmBy EypBx Ez Bz solve
     m_fields.SolvePoissonPsiExmByEypBxEzBz(m_3D_geom, current_N_level);
 
+    // Advance laser slice by 1 step using chi
+    // no MR for laser
+    m_multi_laser.AdvanceSlice(m_fields, m_dt, step);
+
+    if (islice-1 >= m_3D_geom[0].Domain().smallEnd(2)) {
+        m_multi_buffer.get_data(islice-1, m_multi_beam, m_multi_laser, WhichBeamSlice::Next);
+    }
+
+    if (m_N_level > 1) {
+        m_multi_beam.TagByLevel(current_N_level, m_3D_geom, WhichSlice::Next);
+    }
+
     // Bx By solve
     if (m_explicit) {
         for (int lev=0; lev<current_N_level; ++lev) {
             // The algorithm used was derived in
             // [Wang, T. et al. Phys. Rev. Accel. Beams 25, 104603 (2022)],
-            // it is implemented in the WAND-PIC quasistatic PIC code.
+            // it is implemented in the WAND-PIC quasistatic PIC code.#
```

Comment:
```suggestion
            // it is implemented in the WAND-PIC quasistatic PIC code.
```

Just to show that I actually read all of this..